src/core/public/chrome/ui/header/header.tsx

```diff
@@ -266,107 +347,169 @@ class HeaderUI extends Component<Props, State> {
   public renderMenuTrigger() {
     return (
       <EuiHeaderSectionItemButton
-        aria-label="Toggle side navigation"
+        aria-label={i18n.translate('core.ui.chrome.headerGlobalNav.toggleSideNavAriaLabel', {
+          defaultMessage: 'Toggle side navigation',
+        })}
         onClick={() => this.navDrawerRef.current.toggleOpen()}
       >
         <EuiIcon type="apps" size="m" />
       </EuiHeaderSectionItemButton>
     );
   }
 
+  public renderRecentLinks() {
+    return (
+      <EuiNavDrawerGroup
+        listItems={[
+          {
+            label: i18n.translate('core.ui.chrome.sideGlobalNav.viewRecentItemsLabel', {
+              defaultMessage: 'Recently viewed',
+            }),
+            iconType: 'clock',
+            isDisabled: !(this.state.recentlyAccessed.length > 0),
+            flyoutMenu: {
+              title: i18n.translate('core.ui.chrome.sideGlobalNav.viewRecentItemsFlyoutTitle', {
+                defaultMessage: 'Recent items',
+              }),
+              listItems: this.state.recentlyAccessed.map(item => ({
+                label: truncateRecentItemLabel(item.label),
+                title: item.title,
+                'aria-label': item.title,
+                href: item.href,
+                iconType: item.euiIconType,
+              })),
+            },
+          },
+        ]}
+        aria-label={i18n.translate('core.ui.recentLinks.screenReaderLabel', {
+          defaultMessage: 'Recently viewed links, navigation',
+        })}
+      />
+    );
+  }
+
+  public renderNavLinks() {
+    const disableGroupedNavSetting = this.state.navSetting === 'individual';
+    const groupedNavLinks = groupBy(this.state.navLinks, link => link?.category?.label);
+    const { undefined: unknowns, ...allCategorizedLinks } = groupedNavLinks;
+    const { Administration: admin, ...mainCategories } = allCategorizedLinks;
+    const categoryDictionary = getAllCategories(allCategorizedLinks);
+    const orderedCategories = getOrderedCategories(mainCategories, categoryDictionary);
```

Comment:
How does this behave with links not defining a category (`link?.category?` is undefined)? Are they not displayed at all when in `grouped` mode, or does they fallback to a default category?

Reply:
For all products that don't have a category defined, they all get inserted after the last category as top level items. (Continuing to respect their `order` value within that.)